File: tools_C++/src/TestDatabase.cpp
/*
 * TestDatabase.cpp
 *
 *  Created on: Apr 12, 2017
 *      Author: broihier
 */

#include "TestDatabase.h"
#include <iostream>
#include <stdio.h>
#include <execinfo.h>
#include <unistd.h>
#include <stdlib.h>



TestDatabase::TestDatabase(std::string category, std::string project) {
	std::cout << "building test database with category: " << category << " for project " << project << std::endl;
	identification = new Identification(category);
	objective = new Objective(category);
	setup = new Setup(category);
	procedures = new Procedures(category);
	expectedResults = new ExpectedResults(category);
	results = new Results(category);
	cleanup = new Cleanup(category);
	categoryInformation = new CategoryInformation(category);
	std::vector<std::string> requirementsList;
	requirementsList.push_back(project);
	requirements = new Requirements(requirementsList);

}

TestDatabase::~TestDatabase() {

	if (identification) delete identification;
	if (objective) delete objective;
	if (setup) delete setup;
	if (procedures) delete procedures;
	if (expectedResults) delete expectedResults;
	if (results) delete results;
	if (cleanup) delete cleanup;
	if (categoryInformation) delete categoryInformation;
	if (requirements) delete requirements;
}

const std::vector<std::string>& TestDatabase::getIdentificationIds() const{
	return identification->getIdentificationIds();
}

const std::string& TestDatabase::getIdentificationText(const std::string& id) const{
	return identification->getIdentificationText(id);
}

const std::string& TestDatabase::getObjectiveText(const std::string& id) const{
	return objective->getObjectiveText(id);
}
const std::string& TestDatabase::getRequirementsInThisTest(const std::string& id) const{
	return objective->getRequirementsInThisTest(id);
}

const std::string& TestDatabase::getSetupText(const std::string& id) const{
	return setup->getSetupText(id);
}

const std::string& TestDatabase::getProcedureText(const std::string& id) const{
	return procedures->getProcedureText(id);
}

const std::string& TestDatabase::getExpectedResultsText(const std::string& id) const{
	return expectedResults->getExpectedResultsText(id);
}

const std::string& TestDatabase::getResultsText(const std::string& id) const{
	return results->getResultsText(id);
}

const std::string& TestDatabase::getCleanupText(const std::string& id) const{
	return cleanup->getCleanupText(id);
}

const std::string& TestDatabase::getCategoryTitle() const{
	return categoryInformation->getCategoryTitle();
}

const std::string& TestDatabase::getCategoryDescriptionText() const{
	return categoryInformation->getCategoryDescriptionText();
}
const std::string& TestDatabase::getRequirementsText(const std::string& id) const{
	return requirements->getRequirementsText(id);
}

File: tools_java/tools/src/tools/TestDatabase.java
package tools;

import java.util.ArrayList;
/**
 * 
 * @author broihier
 * <pre>
 * Class for creating TestDatabase objects
 * </pre>
 */
public class TestDatabase {
	private CategoryInformation categoryInformation = null;
	private Identification identification = null;
	private Requirements requirements = null;
	private Objective objective = null;
	private Setup setup = null;
	private Procedures procedures = null;
	private ExpectedResults expectedResults = null;
	private Results results = null;
	private Cleanup cleanup = null;
	/** 
	 * Constructor for creating TestDatabase objects
	 * 
	 * @param category category/section/chapter of test procedures
	 * @param project project being tested
	 * 
	 */
	public TestDatabase (String category, String project) {
		categoryInformation = new CategoryInformation(category);
		identification = new Identification(category);
		String [] ids = {project};
		requirements = new Requirements(ids);
		objective = new Objective(category);
		setup = new Setup(category);
		procedures = new Procedures(category);
		expectedResults = new ExpectedResults(category);
		results = new Results(category);
		cleanup = new Cleanup(category);
	}
	/**
	 * Method to get the title of the test category
	 * @return categoryInformation.getCategoryTitle();
	 */
	public String getCategoryTitle() {
		return categoryInformation.getCategoryTitle();
	}
	/**
	 * Method to get the description of the test category
	 * @return categoryInformation.getCategoryDescriptionText();
	 */
	public String getCategoryDescriptionText() {
		return categoryInformation.getCategoryDescriptionText();
	}
	/**
	 * Method to get the list of test IDs for a category
	 * @return identification.getIdentificationIDs();
	 */
	public ArrayList<String> getIdentificationIDs() {
		return identification.getIdentificationIDs();
	}
	/**
	 * Method to get the test case title
	 * @param testID test case identification
	 * @return identification.getIdentificationTest(testID);
	 */
	public String getIdentificationText(String testID) {
		return identification.getIdentificationText(testID);
	}
	/**
	 * Method to get the test objective text
	 * @param testID test case identification
	 * @return objective.getObjectiveText(testID);
	 */
	public String getObjectiveText(String testID) {
		return objective.getObjectiveText(testID);
	}
	/**
	 * Method to get the requirements for a test case
	 * @param testID test case identification
	 * @return objective.getRequirementsInThisTest(testID);
	 */
	public String getRequirementsInThisTest(String testID) {
		return objective.getRequirementsInThisTest(testID);
	}
	/**
	 * Method to get requirements text associated with the requirement ID
	 * @param requirementID requirement ID
	 * @return requirements.getRequirementsText(requirementID);
	 */
	public String getRequirementsText(String requirementID) {
		return requirements.getRequirementsText(requirementID);
	}
	/**
	 * Method to get the text associate with the setup of a test case
	 * @param testID test case identification
	 * @return setup.getSetupText(testID);
	 */
	public String getSetupText(String testID) {
		return setup.getSetupText(testID);
	}
	/**
	 * Method to get the text associated with the procedure steps
	 * @param testID test case identification
	 * @return procedures.getProcedureText(testID);
	 */
	public String getProcedureText(String testID) {
		return procedures.getProcedureText(testID);
	}
	/**
	 * Method to get the expected results text associated with a test case
	 * @param testID test case identification
	 * @return expectedResults.getExpectedResultsTest(testID);
	 */
	public String getExpectedResultsText(String testID) {
		return expectedResults.getExpectedResultsText(testID);
	}
	/**
	 * Method to get the test results of a test case
	 * @param testID test case identification
	 * @return results.getResultsText(testID);
	 */
	public String getResultsText(String testID) {
		return results.getResultsText(testID);
	}
	/**
	 * Method to get the cleanup steps for a test case
	 * @param testID test case identification
	 * @return cleanup.getCleanupText(testID);
	 */
	public String getCleanupText(String testID) {
		return cleanup.getCleanupText(testID);
	}
}
